### ISSUE TITLE
[FEATURE REQUEST] Add func NewServerWithClient() and NewWithClient()?

### ISSUE THREAD

**Issue by Ryalu** [hibiken/asynq#320]
```
// NewWithClient returns a new Client instance given a redis connection option 
// with given `redis.UniversalClient`.
func NewWithClient(cli redis.UniversalClient) *Client {
	return &Client{
		opts: make(map[string][]Option),
		rdb:  rdb.NewRDB(cli),
	}
}

```

```
// NewServerWithClient returns a new Server with given `redis.UniversalClient`
// and server configuration.
func NewServerWithClient(cli redis.UniversalClient, cfg Config) *Server {...}
```

**Comment by hibiken** [comment]
@Ryalu thank you for opening a feature request.

I intentionally didn't add this option because I don't want `asynq` package API to be dependent on a specific `go-redis/redis` package API. For example, if the `NewClient` function takes `github.com/go-redis/redis/v7.UniversalClient` then we would need to bump our major version if we want to update to `github.com/go-redis/redis/v8.UniversalClient`. Or we could potentially use other redis client library and I don't want to make a major version bump because of that. 
In short, I believe the fact that we are using `go-redis/redis` package is an internal implementation detail that shouldn't be exposed in our API. 

Why do you want this as a feature? Is our `RedisConnOpt` missing some field to tune redis client behavior? If so, we can consider adding more options there.

**Comment by Ryalu** [comment]
> @Ryalu thank you for opening a feature request.
> 
> I intentionally didn't add this option because I don't want `asynq` package API to be dependent on a specific `go-redis/redis` package API. For example, if the `NewClient` function takes `github.com/go-redis/redis/v7.UniversalClient` then we would need to bump our major version if we want to update to `github.com/go-redis/redis/v8.UniversalClient`. Or we could potentially use other redis client library and I don't want to make a major version bump because of that.
> In short, I believe the fact that we are using `go-redis/redis` package is an internal implementation detail that shouldn't be exposed in our API.
> 
> Why do you want this as a feature? Is our `RedisConnOpt` missing some field to tune redis client behavior? If so, we can consider adding more options there.

Thanks for replying, I know, I just want to unify my configuration file. 👍

**Comment by hibiken** [comment]
@Ryalu As discussed in #321 , you could implement your own type that satisfies `asynq.RedisConnOpt` interface so that you have full control/access over the redis client. 

Let me know if that works!

**Comment by hibiken** [comment]
Closing this; @Ryalu feel free to reopen if you have any follow up questions!
